src/util/taproot.rs

```diff
@@ -0,0 +1,164 @@
+// Rust Bitcoin Library
+// Written in 2019 by
+//     The rust-bitcoin developers.
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Taproot
+//!
+
+use hashes::{sha256, sha256t};
+
+/// The SHA-256 midstate value for the TapLeaf hash.
```

Comment:
Actually, I'm starting thinking that the whole taproot support in terms of tagged hashes should be part of `bitcoin_hashes` alike other bitcoin-specific Hash160 etc. While this things are taproot-related, they have no other functionality then pure hash computation.

Reply:
Changed my mind after the second thought: https://github.com/rust-bitcoin/rust-bitcoin/pull/259#issuecomment-716446122